samples/boot/oauth2webclient/src/main/java/sample/config/WebClientConfig.java

```diff
@@ -32,8 +35,20 @@
 
 	@Bean
 	WebClient webClient(ClientRegistrationRepository clientRegistrationRepository, OAuth2AuthorizedClientRepository authorizedClientRepository) {
-		ServletOAuth2AuthorizedClientExchangeFilterFunction oauth2 = new ServletOAuth2AuthorizedClientExchangeFilterFunction(clientRegistrationRepository, authorizedClientRepository);
+		OAuth2AuthorizedClientProvider authorizedClientProvider =
```

Comment:
This feels like a lot of code to ask an application to write. Is simplifying this a consideration for another PR?


A reasonable accommodation might be:

```java
OAuth2ClientAuthorizationManager manager = OAuth2ClientAuthorizationManager.withDefaults(clients, authz);
ServletOAuth2AuthorizedClientExchangeFilterFunction oauth2 =
        new ServletOAuth2AuthorizedClientExchangeFilterFunction(manager);
```

Or, if it's an option, leaving the `ServletOAuth2AuthorizedClientExchangeFilterFunction(ClientRegistrationRepository, OAuth2AuthorizedClientRepository)` constructor as un-deprecated would be another solution.

Reply:
In the first example, was that a typo `OAuth2ClientAuthorizationManager`? Did you mean `OAuth2AuthorizedClientManager`? Or is that a new class you are proposing?

For the 2nd example, `ServletOAuth2AuthorizedClientExchangeFilterFunction(ClientRegistrationRepository, OAuth2AuthorizedClientRepository)` will stay deprecated given that `OAuth2AuthorizedClientManager` is now the 1st class citizen for the `ExchangeFilterFunction`.

Reply:
I'm going to change the `WebClientConfig` to be this instead.

```java
@Bean
WebClient webClient(OAuth2AuthorizedClientManager authorizedClientManager) {
	ServletOAuth2AuthorizedClientExchangeFilterFunction oauth2 =
			new ServletOAuth2AuthorizedClientExchangeFilterFunction(authorizedClientManager);
	oauth2.setDefaultOAuth2AuthorizedClient(true);
	return WebClient.builder()
			.apply(oauth2.oauth2Configuration())
			.build();
}

@Bean
OAuth2AuthorizedClientManager authorizedClientManager(ClientRegistrationRepository clientRegistrationRepository,
														OAuth2AuthorizedClientRepository authorizedClientRepository) {
	OAuth2AuthorizedClientProvider authorizedClientProvider =
			OAuth2AuthorizedClientProviderBuilder.withProvider()
					.authorizationCode()
					.refreshToken()
					.clientCredentials()
					.build();
	DefaultOAuth2AuthorizedClientManager authorizedClientManager = new DefaultOAuth2AuthorizedClientManager(
			clientRegistrationRepository, authorizedClientRepository);
	authorizedClientManager.setAuthorizedClientProvider(authorizedClientProvider);

	return authorizedClientManager;
}
```

This simplifies configuring a `WebClient` with the addition of a `OAuth2AuthorizedClientManager` `@Bean`. We can have this `@Bean` auto-configured by Boot but I'm not sure if this is the best option. Reason being is that most user's will not require all grant types to be configured. Thoughts?

Reply:
Sorry, yes, that was a typo - good catch.

I agree that a Boot-provided instance will help out a great deal. I'd expect it to align with other related Boot defaults. For example, doesn't `oauth2Login` get configured by default for the `authorization_code` flow? I would think that the Boot-provided one would do the same, or at least base its decision on the same information.

Reply:
> doesn't `oauth2Login` get configured by default for the authorization_code flow

Technically true, because `oauth2Login` does leverage `authorization_code` flow. But I think you meant `oauth2Client(()`, which by default configures  `authorization_code`.

I still feel that not all users will want to enable all grants by default in this scenario. Of course it wouldn't hurt and they can easily override the default. I'm just not sure if enabling `password` grant by default is a good idea. We'll have to give this some further thought as far as what the default grants will be.

Reply:
Ticket #7115